src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -374,57 +385,93 @@ public void handleSearchContext(SearchContext searchContext, ThreadPool threadPo
             }
 
             PrivilegesEvaluationContext privilegesEvaluationContext = this.dlsFlsBaseContext.getPrivilegesEvaluationContext();
+
             if (privilegesEvaluationContext == null) {
                 return;
             }
 
             DlsFlsProcessedConfig config = this.dlsFlsProcessedConfig.get();
 
-            DlsRestriction dlsRestriction = config.getDocumentPrivileges().getRestriction(privilegesEvaluationContext, index);
+            if (this.isResourceSharingEnabled && OpenSearchSecurityPlugin.getResourceIndices().contains(index)) {
+                CountDownLatch latch = new CountDownLatch(1);
+
+                threadPool.generic()
```

Comment:
See comment at https://github.com/opensearch-project/security/pull/5016/files#r1924173306

Reply:
https://github.com/opensearch-project/security/pull/5016#discussion_r1927298282

Reply:
i think we should align on this asap.